packages/graph/hash_graph/lib/graph/src/store/postgres/mod.rs

```diff
@@ -70,34 +80,115 @@ impl<K> Default for DependencyMap<K> {
+                    // resolved for the full interval anyway.
+                    DependencyStatus::Unresolved(*current_depths, current_interval.clone())
+                } else if old_interval.contains_interval(&new_interval) {
+                    tracing::info!("{old_interval:?} contains {new_interval:?}");
+                    // The dependency is already resolved for the required interval
+                    // old: [-----)
+                    // new:  [---)
                     DependencyStatus::Resolved
+                } else if new_interval.contains_interval(&old_interval)
+                    || new_interval.is_adjacent_to(&old_interval)
+                {
+                    // The dependency is already resolved, but not for the required interval or the
+                    // intervals are adjacent so we only have to resolve the new interval
+                    // old:  [---)   |  [---)
+                    // new: [-----)  |      [---)
+                    DependencyStatus::Unresolved(*current_depths, new_interval)
+                } else if old_interval.overlaps(&new_interval) {
+                    // The time intervals either overlap or are disjoint, so only have to resolve
+                    // the difference of the two intervals.
+                    // old:     [-----)
+                    // new:       [-------)
+                    // resolve:       [---)
+                    let difference = new_interval.difference(old_interval);
+                    // The intervals do overlap and the current interval is not a subset of
+                    // the required interval, so the difference must be a single interval
+                    debug_assert_eq!(difference.len(), 1, "difference must be a single interval");
+
+                    DependencyStatus::Unresolved(
+                        *current_depths,
+                        difference
+                            .into_iter()
+                            .next()
+                            .expect("difference must be a single interval"),
+                    )
+                } else {
+                    // The time intervals are disjoint. Ideally, we only would resolve the new
+                    // interval, but we did not come up with a good way to store the different
+                    // intervals in the dependency map. So we resolve the full interval for now.
+
+                    // This case is expected to happen very rarely, so falling back to this logic
+                    // is fine for now. If it turns out to be a performance bottleneck, we can
+                    // find and implement a better solution. To track this, we log a warning.
+                    tracing::warn!(
+                        vertex_id=?identifier,
+                        ?old_interval,
+                        ?new_interval,
+                        "dependency is resolved for disjoint intervals"
+                    );
```

Comment:
I appreciate you may want to have this here for us to track how frequently it happens but I don't know if I'm comfortable having a tracing statement, with such a high-priority log level (this will appear even if we disable `info` logging) that isn't gated behind a feature or something, and one that is in a potentially majorly hot loop. This _could_ happen tens of thousands of times _per request_.

The log line is also really not self-descriptive, I'd have to actively search for it in the codebase to have any clue what it's about.

Reply:
Do you think it's fine to leave it as a `tracing::debug!` log and create a task to capture this?

Reply:
I think I'd err on the side of removing the log for now and creating a task to figure this out properly. (One we can talk about in refinement and perhaps make a task to establish a metrics collection strategy) 
cc: @thehabbos007